deps/wazuh_testing/wazuh_testing/end_to_end/remote_operations_handler.py

```diff
@@ -0,0 +1,429 @@
+    if wait_is_required:
+        wait_syscollector_and_vuln_scan(host_manager, host, operation_data, current_datetime)
+
+        check_vulnerability_alerts(results, operation_data['check'], current_datetime, host_manager, host,
+                                   {'from': package_data_from, 'to': package_data_to} , operation='update')
+        return {
+                f"{host}": results
+            }
+
+
+def launch_remote_sequential_operation_on_agent(agent: str, task_list: List[Dict], host_manager: HostManager):
+    """
+    Launch sequential remote operations on an agent.
+
+    Args:
+        agent (str): The target agent on which to perform the operations.
+        task_list (list): List of dictionaries containing operation details.
+        host_manager (HostManager): An instance of the HostManager class containing information about hosts.
+    """
+    # Convert datetime to Unix timestamp (integer)
+    timestamp = datetime.utcnow().isoformat()
+
+    if task_list:
+        for task in task_list:
+            operation = task['operation']
+            if operation in locals():
+                locals()[operation](agent, task, host_manager, timestamp)
+
+
+def launch_remote_operation(host: str, operation_data: Dict[str, Dict], host_manager: HostManager):
+    operation = operation_data['operation']
+    if operation in globals():
+        operation_result = globals()[operation](host, operation_data, host_manager)
+        logging.critical(f"Operation result: {operation_result}")
+        return operation_result
+    else:
+        raise ValueError(f"Operation {operation} not recognized")
+
+
+def launch_parallel_operations(task_list: List[Dict], host_manager: HostManager, target_to_ignore: List[str] = []):
+    """
+    Launch parallel remote operations on multiple hosts.
+
+    Args:
+        task_list (list): List of dictionaries containing operation details.
+        host_manager (HostManager): An instance of the HostManager class containing information about hosts.
+    """
+    results = {}
+
+    if target_to_ignore:
+        for target in results:
+            results[target]['checks']['all_successfull'] = False
+
+    def launch_and_store_result(args):
+        host, task, manager = args
+        result = launch_remote_operation(host, task, manager)
+        results.update(result)
+
+    with ThreadPoolExecutor() as executor:
+        # Submit tasks asynchronously
+        futures = []
+        for task in task_list:
+            hosts_target = host_manager.get_group_hosts(task['target'])
+            if target_to_ignore:
+                hosts_target = [host for host in hosts_target if host not in target_to_ignore]
+
+            logging.info("Hosts target after removing ignored targets: {}".format(hosts_target))
+
+            for host in hosts_target:
+                futures.append(executor.submit(launch_and_store_result, (host, task, host_manager)))
```

Comment:
It is more clear to indicate on which hosts to run the tasks, than on which not, but if there is no other approach, we can leave it like this.

Reply:
Agree. We should change this behavior in future Functionality testing Design